android/src/main/kotlin/com/google/maps/flutter/navigation/GoogleMapsNavigationView.kt

```diff
@@ -731,360 +232,6 @@ internal constructor(
-    invalidateViewAfterMapLoad()
-    var error: Throwable? = null
-    polylines.forEach {
-      findPolylineController(it.polylineId)?.let { controller ->
-        controller.remove()
-        _polylines.remove(controller)
-      }
-        ?: run {
-          error =
-            FlutterError("polylineNotFound", "Failed to remove polyline with id ${it.polylineId}")
-        }
-    }
-    error?.let { throw error as Throwable }
-  }
-
-  fun clearPolylines() {
-    invalidateViewAfterMapLoad()
-    _polylines.forEach { controller -> controller.remove() }
-    _polylines.clear()
-  }
-
-  fun getCircles(): List<CircleDto> {
-    val density = Resources.getSystem().displayMetrics.density
-    return _circles.map {
-      CircleDto(it.circleId, Convert.circleToCircleOptions(it.circle, density))
-    }
-  }
-
-  fun addCircles(circles: List<CircleDto>): List<CircleDto> {
-    val density = Resources.getSystem().displayMetrics.density
-    invalidateViewAfterMapLoad()
-    val result = mutableListOf<CircleDto>()
-    circles.forEach {
-      val builder = CircleBuilder()
-      Convert.sinkCircleOptions(it.options, builder, density)
-      val options = builder.build()
-      val circle = getMap().addCircle(options)
-      if (circle != null) {
-        val controller = CircleController(circle, it.circleId)
-        _circles.add(controller)
-        result.add(it)
-      }
-    }
-    return result
-  }
-
-  fun updateCircles(circles: List<CircleDto>): List<CircleDto> {
-    val density = Resources.getSystem().displayMetrics.density
-    invalidateViewAfterMapLoad()
-    val result = mutableListOf<CircleDto>()
-    var error: Throwable? = null
-    circles.forEach {
-      findCircleController(it.circleId)?.let { controller ->
-        Convert.sinkCircleOptions(it.options, controller, density)
-        result.add(it)
-      }
-        ?: run {
-          error = FlutterError("circleNotFound", "Failed to update circle with id ${it.circleId}")
-        }
-    }
-    error?.let { throw error as Throwable }
-    return result
-  }
-
-  fun removeCircles(circles: List<CircleDto>) {
-    invalidateViewAfterMapLoad()
-    var error: Throwable? = null
-    circles.forEach {
-      findCircleController(it.circleId)?.let { controller ->
-        controller.remove()
-        _circles.remove(controller)
-      }
-        ?: run {
-          error = FlutterError("circleNotFound", "Failed to remove circle with id ${it.circleId}")
-        }
-    }
-    error?.let { throw error as Throwable }
-  }
-
-  fun clearCircles() {
-    invalidateViewAfterMapLoad()
-    _circles.forEach { controller -> controller.remove() }
-    _circles.clear()
-  }
-
-  fun setConsumeMyLocationButtonClickEventsEnabled(enabled: Boolean) {
-    _consumeMyLocationButtonClickEventsEnabled = enabled
-  }
-
-  fun isConsumeMyLocationButtonClickEventsEnabled(): Boolean {
-    return _consumeMyLocationButtonClickEventsEnabled
-  }
-
   fun registerOnCameraChangedListener() {
```

Comment:
Should this be part of the base class.
Both MapView and NavigationView should be able to listeb onCameraChanged events?